src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -2294,6 +2312,28 @@ def _ignore_handler(self, *_):
         '''
         return None
 
+    def _include_handler(self, node, parent):
+        '''
+        Handler for Fortran INCLUDE statements. Since these are not supported
+        by the PSyIR it simply raises an error.
+
+        :param node: node in fparser2 tree.
+        :type node: :py:class:`fparser.two.Fortran2003.Include_Stmt`
+        :param parent: parent node of the PSyIR node we are constructing.
+        :type parent: :py:class:`psyclone.psyir.nodes.Schedule`
+
+        :raises GenerationError: as INCLUDE statements must be handled by \
+                                 the parser.
+        '''
+        config = Config.get()
+        raise GenerationError(
+            f"Fortran INCLUDE statements are not supported but found an "
+            f"include of file '{node.children[0].string}' while processing "
+            f"routine '{parent.name}'. This file must be made available to "
```

Comment:
Since "include statements" can be anywhere, can we guarantee that parent is a routine and has a name?

Reply:
No we couldn't - good spot. I've improved this and extended the tests.